src/openrct2/peep/peep.c

```diff
@@ -9823,10 +9823,8 @@ static sint32 guest_path_find_leaving_park(rct_peep *peep, rct_map_element *map_
 	rct2_peep_spawn* peepSpawn = &gPeepSpawns[0];
 	// Peeps for whatever reason return to their original spawn point
 	// this in future should look for the nearest.
-	if (peep->sprite_index & 1 && gPeepSpawns[1].x != PEEP_SPAWN_UNDEFINED) {
-		peepSpawn++;
-	}
-
+	get_random_peep_spawn(peepSpawn);
```

Comment:
The comment above needs editing. I don't think it was ever true. Just a thought but does this code get called multiple times whilst leaving the park as you would cause the peep to keep changing destination.